docs/source/getting_started/benchmarks/cifar10_benchmark.py

```diff
@@ -0,0 +1,365 @@
+        loss_2 = self.criterion(*self.resnet_moco(x1, x0))
+        loss = 0.5 * (loss_1 + loss_2)
+        self.log('train_loss_ssl', loss)
+        return loss
+
+    def configure_optimizers(self):
+        optim = torch.optim.SGD(self.resnet_moco.parameters(), lr=6e-2,
+                                momentum=0.9, weight_decay=5e-4)
+        scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optim, max_epochs)
+        return [optim], [scheduler]
+
+
+class SimCLRModel(BenchmarkModule):
+    def __init__(self, dataloader_kNN):
+        super().__init__(dataloader_kNN)
+        # create a ResNet backbone and remove the classification head
+        resnet = lightly.models.ResNetGenerator('resnet-18')
+        self.backbone = nn.Sequential(
+            *list(resnet.children())[:-1],
+            nn.AdaptiveAvgPool2d(1),
+        )
+        # create a simclr model based on ResNet
+        self.resnet_simclr = \
+            lightly.models.SimCLR(self.backbone, num_ftrs=512)
+        self.criterion = lightly.loss.NTXentLoss()
+            
+    def forward(self, x):
+        self.resnet_simclr(x)
+
+    def training_step(self, batch, batch_idx):
+        (x0, x1), _, _ = batch
+        x0, x1 = self.resnet_simclr(x0, x1)
+        loss = self.criterion(x0, x1)
+        self.log('train_loss_ssl', loss)
+        return loss
+
+    def configure_optimizers(self):
+        optim = torch.optim.SGD(self.resnet_simclr.parameters(), lr=6e-2,
+                                momentum=0.9, weight_decay=5e-4)
+        scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optim, max_epochs)
+        return [optim], [scheduler]
+
+
+class SimSiamModel(BenchmarkModule):
+    def __init__(self, dataloader_kNN):
+        super().__init__(dataloader_kNN)
+        # create a ResNet backbone and remove the classification head
+        resnet = lightly.models.ResNetGenerator('resnet-18')
+        self.backbone = nn.Sequential(
+            *list(resnet.children())[:-1],
+            nn.AdaptiveAvgPool2d(1),
+        )
+        # create a simsiam model based on ResNet
+        self.resnet_simsiam = \
+            lightly.models.SimSiam(self.backbone, num_ftrs=512, num_mlp_layers=2)
+        self.criterion = lightly.loss.SymNegCosineSimilarityLoss()
+            
+    def forward(self, x):
+        self.resnet_simsiam(x)
+
+    def training_step(self, batch, batch_idx):
+        (x0, x1), _, _ = batch
+        x0, x1 = self.resnet_simsiam(x0, x1)
+        loss = self.criterion(x0, x1)
+        self.log('train_loss_ssl', loss)
+        return loss
+
+    def configure_optimizers(self):
+        optim = torch.optim.SGD(self.resnet_simsiam.parameters(), lr=6e-2,
+                                momentum=0.9, weight_decay=5e-4)
+        scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optim, max_epochs)
+        return [optim], [scheduler]
+
+
+model_names = ['SimCLR_512', 'SimSiam_512']
+models = [SimCLRModel, SimSiamModel]
```

Comment:
You're using batch sizes `128` and `512` yet the model names say `512`. To make the scripts safe, should we not set the model names dynamically? Also, where is MoCo? I would add it to the list of models.

Reply:
done